python-avd/pyavd/_schema/models/avd_indexed_list.py

```diff
@@ -0,0 +1,179 @@
+        return cls(cls_items)
+
+    def __init__(self, items: Iterable[T_AvdModel] | UndefinedType = Undefined) -> None:
+        if isinstance(items, UndefinedType):
+            self._items = {}
+        else:
+            self._items = {getattr(item, self._primary_key): item for item in items}
+
+    def __repr__(self) -> str:
+        """Returns a repr with all the items including any nested models."""
+        cls_name = self.__class__.__name__
+        attrs = [f"{item!r}" for item in (self._items or ())]
+        return f"<{cls_name}([{', '.join(attrs)}])>"
+
+    def __bool__(self) -> bool:
+        """Boolean check on the class to quickly determine if any items are set."""
+        return bool(self._items)
+
+    def __len__(self) -> int:
+        return self._items.__len__()
+
+    def __contains__(self, key: T_PrimaryKey) -> bool:
+        return self._items.__contains__(key)
+
+    def __iter__(self) -> Iterator[T_AvdModel]:
+        return self._items.values().__iter__()
+
+    def __getitem__(self, key: T_PrimaryKey) -> T_AvdModel:
+        return self._items[key]
+
+    def __setitem__(self, key: T_PrimaryKey, value: T_AvdModel) -> None:
+        self._items[key] = value
+
+    def get(self, key: T_PrimaryKey, default: T | UndefinedType = Undefined) -> T_AvdModel | T | UndefinedType:
+        return self._items.get(key, default)
+
+    def items(self) -> Iterable[tuple[T_PrimaryKey, T_AvdModel]]:
+        return self._items.items()
+
+    def keys(self) -> Iterable[T_PrimaryKey]:
+        return self._items.keys()
+
+    def values(self) -> Iterable[T_AvdModel]:
+        return self._items.values()
+
+    def append(self, item: T_AvdModel) -> None:
+        self._items[getattr(item, self._primary_key)] = item
+
+    def extend(self, items: Iterable[T_AvdModel]) -> None:
+        self._items.update({getattr(item, self._primary_key): item for item in items})
+
+    def _as_list(self, include_default_values: bool = False) -> list[dict]:
+        """Returns a list with all the data from this model and any nested models."""
+        return [item._as_dict(include_default_values=include_default_values) for item in self._items.values()]
+
+    def _natural_sorted(self, ignore_case: bool = True) -> Self:
+        """Returns an iterator yielding the items natural sorted by primary key."""
+
+        def convert(text: str) -> int | str:
+            if text.isdigit():
+                return int(text)
+            return text.lower() if ignore_case else text
+
+        def key(value: T_AvdModel) -> list[int | str]:
+            primary_key = getattr(value, self._primary_key)
+            return [convert(c) for c in re.split(NATURAL_SORT_PATTERN, str(primary_key))]
+
+        cls = type(self)
+        return cls(sorted(self.values(), key=key))
+
+    def _deepmerge(self, other: Self, list_merge: Literal["append", "replace"] = "append") -> None:
+        """Update instance by deepmerging the other instance in."""
+        cls = type(self)
```

Comment:
Could use this everywhere instead
```suggestion
        cls = self.__class__
```

Reply:
I would say type() is better. Same as we do with bool() and int() and str().